src/rt/sched/threadpool.h

```diff
@@ -238,6 +250,31 @@ namespace verona::rt
       if (local() != nullptr)
         yield_my_turn();
     }
+
+    static void thread_finished()
+    {
+      auto& sched = get();
+      auto me = local();
+      assert(me->systematic_state == SystematicState::Active);
+      me->systematic_state = SystematicState::Finished;
+
+      assert(sched.running_thread == me);
+
+      // Confirm at least one other thread is running,
+      auto curr = me;
+      while (curr->systematic_state != SystematicState::Active)
+      {
+        curr = curr->next;
+        if (curr == me)
+        {
+          // No threads left
+          running_thread = nullptr;
```

Comment:
```suggestion
          sched.running_thread = nullptr;
```

Reply:
It's always the thing you rebase in at the last minute you get wrong ;-)